fails to build in ghc-9.6 stackage nightly

```
       Building library for cron-0.7.0..                                                                                [282/157179]
       [10 of 12] Compiling System.Cron.Schedule                                                                                    
                                                                                                                                    
       /var/stackage/work/unpack-dir/unpacked/cron-0.7.0-a644a72eef6377a56613b423ddc438ab324361759876a64cec8ed479374a36ff/src/System
/Cron/Schedule.hs:142:30: error: [GHC-88464]
           Variable not in scope: forever :: IO b0 -> IO ()                                                                         
           |                     
       142 | forkJob (Job s a) = forkIO $ forever $ do                                                                              
           |                              ^^^^^^^                                                                                   
                                 
       /var/stackage/work/unpack-dir/unpacked/cron-0.7.0-a644a72eef6377a56613b423ddc438ab324361759876a64cec8ed479374a36ff/src/System
/Cron/Schedule.hs:145:13: error: [GHC-88464]
           Variable not in scope: when :: Bool -> t0 -> IO b0                                                                       
           |                     
       145 |             when (scheduleMatches s timeAt) (void $ forkIO a)
           |             ^^^^
                                 
       /var/stackage/work/unpack-dir/unpacked/cron-0.7.0-a644a72eef6377a56613b423ddc438ab324361759876a64cec8ed479374a36ff/src/System
/Cron/Schedule.hs:145:46: error: [GHC-88464]
           Variable not in scope: void :: IO ThreadId -> t0                                                                         
           |                     
       145 |             when (scheduleMatches s timeAt) (void $ forkIO a)
           |                                              ^^^^
```

--- a/cron.cabal
+++ b/cron.cabal
@@ -58,7 +58,7 @@ library
   default-language:    Haskell2010
   build-depends:     base            >= 4 && < 5
                    , attoparsec      >= 0.10
-                   , text            >= 0.11 && < 2
+                   , text            >= 0.11 && < 3
                    , time            >= 1.4
                    , old-locale      >= 1.0
                    , mtl             >= 2.0.1

--- a/src/System/Cron/Schedule.hs
+++ b/src/System/Cron/Schedule.hs
@@ -49,6 +49,7 @@ module System.Cron.Schedule
 import           Control.Applicative
 #endif
 import           Control.Concurrent
+import           Control.Monad (forever, when, void)
 import           Control.Monad.Except
 import           Control.Monad.Identity
 import           Control.Monad.State

